tests/unit/test_server.py

```diff
@@ -40,11 +41,56 @@ def test_prometheus_server_reload_configuration_success(self):
         self.assertTrue(self.prometheus.reload_configuration())
 
     @responses.activate
-    def test_prometheus_server_reload_configuration_failure(self):
+    def test_prometheus_server_without_route_prefix_reload_configuration_failure(self):
+        self.prometheus = Prometheus("localhost", 9090)
+
         responses.add(
             responses.POST,
             "http://localhost:9090/-/reload",
             status=500,
         )
 
         self.assertFalse(self.prometheus.reload_configuration())
+
+    @responses.activate
+    def test_prometheus_server_with_route_prefix_returns_valid_data(self):
```

Comment:
Since most of these tests involve mocking I added a note to myself in [Prometheus issue 184](https://github.com/canonical/prometheus-k8s-operator/issues/184) as a reminder to create integration tests for ingress+Prometheus. If you have a deeper insight here on what the test case should look like kindly do refine it in the issue tracker.